src/shared/components/app/navbar.tsx

```diff
@@ -47,13 +49,18 @@ function handleLogOut(i: Navbar) {
 }
 
 export class Navbar extends Component<NavbarProps, NavbarState> {
+  private userAgent: UAParser.IResult = (
+    this.context.router.staticContext as RouterContext
+  ).userAgent;
+
   state: NavbarState = {
     unreadInboxCountRes: { state: "empty" },
     unreadReportCountRes: { state: "empty" },
     unreadApplicationCountRes: { state: "empty" },
   };
   collapseButtonRef = createRef<HTMLButtonElement>();
   mobileMenuRef = createRef<HTMLDivElement>();
+  isMobileUA: boolean = isMobile(this.userAgent);
```

Comment:
The type should be able to be inferred correctly without the explicit boolean type.

Reply:
right, I can remove it